extensions/typescript-language-features/src/typescriptServiceClient.ts

```diff
@@ -346,6 +348,7 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 			}
 			this.serviceExited(!this.isRestarting);
 			this.isRestarting = false;
+			rimraf.sync(electron.getRootTempDir());
```

Comment:
This will be invoked when the server restarts, which I don't think is correct since there is only a single temp dir for the entire extension. You can test this using the `Restart TS Server` command with a js/ts file open. I believe it will error when it tries created a new cancellation file in the deleted temp dir

Two options:

* Create a temp dir for each server process
* Move the delete to the extension's deactivate method

Reply:
I don't believe this will error since `getRootTempDir()` creates the directory if it does not exist. Therefore when it attempts to make a cancellation file it calls this function and then joins the path with the file name effectively preventing the directory from never not existing when being written to. I will test it out though just to be sure

Reply:
I would not rely on that behavior. That code was added to fix a bug: f264e8f33bfe5eebd3526844fa3519be88ea7a1a

The intent of this function is temp direction is created once and exists over the lifespan on the extension

Reply:
Moved folder removal to deactivate method.